src/librustdoc/clean/mod.rs

```diff
@@ -2568,8 +2568,12 @@ fn resolve_type(cx: &DocContext,
     debug!("resolve_type({:?},{:?})", path, id);
     let tcx = match cx.tcx_opt() {
         Some(tcx) => tcx,
-        // If we're extracting tests, this return value doesn't matter.
-        None => return Primitive(Bool),
+        // If we're extracting tests, this return value's accuracy is not important, all we
+        // want is a string representation to help people figure out what doctests are failing.
+        None => {
+            let did = DefId::local(DefIndex::from_u32(0));
+            return ResolvedPath { path: path, typarams: None, did: did, is_generic: false }
```

Comment:
The majority of types, mainly anything more complicated than a primitive, would become `Primitive(Bool)`, which renders as `bool`, this still skips the type resolve and uses the path we've currently got which is good enough for the test outputs as it should match what the user actually wrote on their impl. (and make things easy to find?)